src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -41,8 +41,19 @@ namespace ripple {
 static std::optional<uint256>
 parseIndex(Json::Value const& params, Json::Value& jvResult)
 {
+    std::string const index = params.asString();
```

Comment:
I think I may have miscommunicated my suggestion - I was suggesting that 
`amendments: "01234...."` should only accept the actual `amendments` key, and not other values.

Reply:
Only the correct key will give you the `Amendments` object. Anything else will give you `objectNotFound` or `unexpectedLedgerType`.

Reply:
I disagree with supporting these strings. Currently `index` only supports uint256s and nothing else, and this feels like a breaking change to that API to me.

Reply:
Yes, I see your point. I have counterpoints
1. _Technically_, `index` only supports strings, which are then parsed into `uint256`. Since we're parsing anyway, why not parse a few special-case strings into other `uint256`s?
2. I was modelling the behavior on the `ledger_index` parameter, which normally takes numbers, but can also take the special-case values of "open", "closed", and "validated".
3. I usually think of breaking changes as things that move or remove an input or output value, such that an existing integration that's expecting something to be in a certain place can't find it anymore. This is more like an addition. It can be completely ignored, and nothing will suffer for it.

I'm willing to take it out or gate it on API version if you still think it's breaking. Thoughts?